src/tests.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * tests.c
+ *
+ * Created: 27.11.2018 16:10:19
+ *  Author: jr707964
+ */
+
+#include <stdio.h>
+#include "tests.h"
+#include "main.h"
+#include "api.h"
+#include "compat.h"
+#include "errors.h"
+
+void program_uart_heater_test(void) {
+  API api;
+  float f1;
+  float f2;
+  int h;
```

Comment:
While I'm not a huge fan of short variable names I'm fine with introducing them to the codebase when they are still somewhat descriptive.  
e.g. `f1` and `f2` don't really say anything about their purpose, how about `t1` and `t2` (for temperature) instead if you want to keep them short?